[docs] Instruct user to check Firebase API Key restrictions (#12253)
expo/expo: docs/pages/push-notifications/using-fcm.md

4. In your app.json, add an `android.googleServicesFile` field with the relative path to the `google-services.json` file you just downloaded. If you placed it in the root directory, this will probably look like

  ```javascript
  {
    ...
    "android": {
      "googleServicesFile": "./google-services.json",
      ...
    }
  }
  ```

> 5.  Confirm that the API key in the `google-services.json` file has the correct restrictions in the [Google Cloud Platform API Credentials console](https://console.cloud.google.com/apis/credentials). Firebase requires the API key to have access to both `Firebase Cloud Messaging API` and `Firebase Installations API` for push notifications to work correctly. The API key in the `google-services.json` file can be found under the `client.api_key.current_key` field, e.g.

Reviewer: ```suggestion
5.  Confirm that your API key in `google-services.json` has the correct "API restrictions" in the [Google Cloud Platform API Credentials console](https://console.cloud.google.com/apis/credentials). For push notifications to work correctly, Firebase requires the API key to either be unrestricted (the key can call any API), or have access to both `Firebase Cloud Messaging API` and `Firebase Installations API`. The API key can be found under the `client.api_key.current_key` field in `google-services.json`.
```

  ```javascript
  {
    ...
    "client": [
      {
        "api_key": [
          {
            "current_key" "<your Google Cloud Platform API key>",
          }
        ],
      }
    ]

>   }

Reviewer: I think since we already state it can be found under `client.api_key.current_key`, this might be a little extraneous

Author: I'd prefer to keep this in, unless you have major objections. I am personally a more visual reader/learner, and the snippets are helpful for me.

  ```

>   **Note:** Firebase will create an API key in the Google Cloud Platform console with a name like `Android key (auto created by Firebase)`. **This is not always the same key as the one found in `google-services.json`. Always confirm your key and associated restrictions in the Google Cloud Platform console.**

Reviewer: ```suggestion
>  **Note:** Firebase will create an API key in the Google Cloud Platform console with a name like `Android key (auto created by Firebase)`. **This is not always the same key as the one found in `google-services.json`. Always confirm your key and associated restrictions in the Google Cloud Platform console.**
```
